### ISSUE TITLE
[Accessibility] : FD-TABLE - F2 button should be supported for focusable table row when [allCellsFocusable]="false"

### ISSUE THREAD

**Issue by harshitha-s-kanchan** [SAP/fundamental-ngx#11546]
#### Is this a bug, enhancement, or feature request?
bug / enhancement
#### Describe your proposal.
In a scenario where the table row needs to be focused (by adding `[focusable]="true" on <tr>`) and each cell should not be focused (by adding `[allCellsFocusable]="false" on <table>`) and there are interactive elements in few columns (Below is the screenshot of the example)
<img width="505" alt="image" src="https://github.com/SAP/fundamental-ngx/assets/149040487/b7b39784-b3b4-47aa-990d-d276c7d902f5">

The screen reader reads out  Eg : 'Row: 3 of 11, use F2 button to dive in and focus list's item', but F2 key is not supported when the row is focused and the interactive elements are not accessible via keyboard. 

F2 key only works when [allCellsFocusable]="true" is added on the table. Enabling [allCellsFocusable]="true" will create two tab stops to access the checkboxes or links within the row. Additional tab stops is an accessibility issue, as focusing on the cells is irrelevant to the user in this case

The F2 button should be enabled on the row focus if there are any interactive elements inside the row, irrespective whether cells are focusable or not.

#### Can you handle that on the application side
No, Since F2 is handled inside the fd-table component

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)

Component - `<fd-table>`
Angular - 15.2.9
NGX - 0.43.21 (Issue also present in latest version)
Screen reader - JAWS 2023

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
Example Link - https://stackblitz.com/edit/5tj8lk?file=src%2Fapp%2Ftable-focusable-example.component.html

Steps: 
1. Focus on the row 
2. Click F2 key (Doesn't focus inside the row)

#### Please provide stackblitz example(s).
https://stackblitz.com/edit/5tj8lk?file=src%2Fapp%2Ftable-focusable-example.component.html

**Comment by droshev** [comment]
@mikerodonnell89  will follow up with @g-cheishvili to understand the next steps

**Comment by droshev** [comment]
@harshitha-s-kanchan do you think you can give us more context?

**Comment by droshev** [comment]
no activities => closing the issue
